tools/wkt2svg.rs

```diff
@@ -270,8 +458,31 @@ fn main() {
     let reader = get_input_reader(&args.input).unwrap();
     // Can't lazily convert to SVG because we have to know the whole collection's bounding box to
     // know how to scale.
-    let geometries = read_geometries(reader, &args.input_format);
-    let geometries: Vec<_> = flatten_nested_geometries(geometries).collect();
+    let geometries = read_wkt_geometries_and_styles(reader);
+
+    // Flatten any MULTI or GEOMETRYCOLLECTION geometries
+    let geometries: Vec<_> = geometries.collect();
+    // [g, g, g, s, g, s, s, g] => [g, g, g, s], [g, s], [s], [g]
+    let geometries_ending_with_styles =
+        geometries.split_inclusive(|x| matches!(x, GeometryAndStyle::Style(_)));
+    let flattened = geometries_ending_with_styles.map(|slice| {
+        if matches!(slice.last(), Some(GeometryAndStyle::Style(_))) {
+            let (style, geometries) = slice.split_last().unwrap();
+            // Have to convert to geo_types to flatten
+            let flattened = flatten_nested_geometries(geometries.iter().map(|g| g.clone().into()));
+            // But then need to convert back to add the style back in
+            let flattened = flattened.map(|g| g.into());
+            let mut flattened: Vec<GeometryAndStyle> = flattened.collect();
+            flattened.push(style.clone());
+            flattened
+        } else {
+            let flattened = flatten_nested_geometries(slice.iter().map(|g| g.clone().into()));
+            let flattened = flattened.map(|g| g.into());
+            flattened.collect()
+        }
+    });
+    let geometries: Vec<_> = flattened.flatten().collect();
+
```

Comment:
This... is unfortunate. I _hate_ it, but I'd rather do this than let the new `GeometryAndStyle` type infect everything else (e.g., `flatten_nested_geometries()`)